### DISCUSSION TITLE
Enable adapter-static to connect with adapter-node backend by simply supplying URL

### DISCUSSION THREAD

**Discussion by bluepuma77** [sveltejs/kit#12161] [Q&A]
### Describe the problem

I would like to create a mobile app from of my SvelteKit web application. CapacitorJS enables this, it can package a static SvelteKit app into a mobile application ([Ionic CapacitorJS blog](https://ionic.io/blog/pwas-using-sveltekit-and-ionic), [svelte-capacitor-recipe](https://github.com/bersling/svelte-capacitor-recipe), [sveltekit-capacitor-starter](https://github.com/khromov/sveltekit-capacitor-starter), [capkit](https://github.com/Hugos68/capkit)), available on an App Store.

My current understanding is that an adapter-static build does not include the automatic fetch functionality that a regular adapter-node SvelteKit web app has included. So for every SvelteKit page I have to manually generate the fetch logic.

### Describe the proposed solution

It would be great if adapter-static could be used to build a static SvelteKit web application, that can still connect to a regular SvelteKit (adapter-node) backend by simply supplying the backend URL during build or runtime.

Further enhancement would be to modify this fetch functionality centrally to supply a http header like access token.

### Alternatives considered

Manually generate fetch logic for every page implemented.

### Importance

would make my life easier

**Comment by bluepuma77** [comment]
@Rich-Harris, @benmccann, @dummdidumm, @Conduitry:  
Sorry for mentioning, would love to get a quick comment if this is already possible, something desirable or out of scope.

**Comment by Conduitry** [comment]
Please don't ping maintainers.

I'm really not sure what this is asking for. `+page.js` and `load` _do_ work in adapter-static apps. Have you tried it? If you want to prevent pages from getting rendered as HTML during build so that they'll instead get rendered by the browser at runtime, you can also do that.

**Reply by bluepuma77** [reply]
With SvelteKit the page can automatically load data from the server, without writing any `fetch` code:
```
<!-- +page.svelte -->
<script>
	export let data;
</script>
<h1>{data.post.title}</h1>
```
```
// +page.server.ts
export const load = async ({ params }) => {
	return {
		post: { title: `Title for ${params.slug} goes here` }
	}
};
```

When using SvelteKit with Capacitor, I need to generate a `static` version that runs independently on mobile. But I want the `static` version to still have the automatic fetch-from-server functionality, so I don't have to manually write a fetch for every page. All it should need is the server URL to provide the same behaviour.

**Reply by bluepuma77** [reply]
@Conduitry Can you answer this?

**Reply by PaulRopel** [reply]
@bluepuma77 do you have any new insides on this? I am currently investigating the same problem. How to reuse the .server bits when using the static adapter
